spec/lib/spotlight/dor/indexer_spec.rb

```diff
@@ -498,4 +498,87 @@
     end # each
   end # add_folder_name
   # rubocop:enable Metrics/LineLength
+
+  # rubocop:disable Metrics/LineLength
+  describe '#add_object_full_text' do
+    let(:expected_text) do
+      'SOME full text string that is returned from the server'
+    end
+    let(:full_file_path) do
+      'https://stacks.stanford.edu/file/oo000oo0000/oo000oo0000.txt'
+    end
+    let(:public_xml_with_feigenbaum_full_text) do
```

Comment:
Thanks jessie.  This was my first experience with either, useful to know.

From: Github Notifications <notifications@github.com<mailto:notifications@github.com>>
Reply-To: sul-dlss/spotlight-dor-resources <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, November 3, 2015 at 6:49 PM
To: sul-dlss/spotlight-dor-resources <spotlight-dor-resources@noreply.github.com<mailto:spotlight-dor-resources@noreply.github.com>>
Cc: Peter Mangiafico <petucket@stanford.edu<mailto:petucket@stanford.edu>>
Subject: Re: [spotlight-dor-resources] add full text indexing and tests (#26)


In spec/lib/spotlight/dor/indexer_spec.rb<https://github.com/sul-dlss/spotlight-dor-resources/pull/26#discussion_r43837639>:

> @@ -498,4 +498,87 @@
>      end # each
>    end # add_folder_name
>    # rubocop:enable Metrics/LineLength
> +
> +  # rubocop:disable Metrics/LineLength
> +  describe '#add_object_full_text' do
> +    let(:expected_text) do
> +      'SOME full text string that is returned from the server'
> +    end
> +    let(:full_file_path) do
> +      'https://stacks.stanford.edu/file/oo000oo0000/oo000oo0000.txt'
> +    end
> +    let(:public_xml_with_feigenbaum_full_text) do


Not to chime-in unnecessarily but I just wanted to clear something up about let vs let!. The primary difference between the bang version (let!) is that it the value is evaluated immediately instead of lazily (i.e. let is evaluated when the variable is invoked where let! is evaluated immediately when set).

The availability of the value in the let/let! is more about the scope in which it is defined.

https://www.relishapp.com/rspec/rspec-core/v/2-5/docs/helper-methods/let-and-let

—
Reply to this email directly or view it on GitHub<https://github.com/sul-dlss/spotlight-dor-resources/pull/26/files#r43837639>.